projects/interfaces/main.go

```diff
@@ -0,0 +1,91 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+)
+
+type OurByteBuffer struct {
+	myBuff []byte
+}
+
+// Is there a maximum about of data an OurByteBuffer can store?
+// There is no maximum as far as our memory doesn't run out.
+// As far as I learned, buffer (bytes.buffer) can grow magically and always is the same buffer.
+// But OurByteBuffer created as a slice bytes and can grow but always not the same slice bytes.
```

Comment:
Can you talk through what you think the difference is between "same buffer" and "not the same slice bytes"?

In terms of memory, what do you think is happening in each case?

Reply:
Sorry I learned wrong. Before I thought that buffer is a temporary slice in memory that the location is always same and can grow until memory run out.
Update: Buffer and slice are doing same action when they grow. Copying, updating the header.